src/features/device-logs/lib/backends/loki.ts

```diff
@@ -210,11 +206,23 @@ export class LokiBackend implements DeviceLogsBackend {
 		const pushRequest = new PushRequest();
 		pushRequest.setStreamsList(streams);
 		const startAt = Date.now();
-		return this.pusher
-			.push(pushRequest, createOrgIdMetadata(String(appId)), {
-				deadline: Date.now() + PUSH_TIMEOUT,
-			})
-			.finally(() => updateLokiPushDurationHistogram(Date.now() - startAt));
+		return new Promise<any>((resolve, reject) => {
+			this.pusher.push(
+				pushRequest,
+				createOrgIdMetadata(String(appId)),
+				{
+					deadline: Date.now() + PUSH_TIMEOUT,
+				},
+				(err, response) => {
+					updateLokiPushDurationHistogram(Date.now() - startAt);
```

Comment:
Can you move this back into a `.finally` on the promise itself, it's best to have the minimum in a callback so that you get back to promise land ASAP and avoid issues with error handling in callbacks